test/unit/compiler/passes/merge-character-classes.spec.js

```diff
@@ -0,0 +1,38 @@
+"use strict";
+
+const chai = require("chai");
+const helpers = require("./helpers");
+const pass = require("../../../../lib/compiler/passes/merge-character-classes");
+
+chai.use(helpers);
+
+const expect = chai.expect;
+
+describe("compiler pass |mergeCharacterClasses|", () => {
+  it("Merges single character literals, class, and ref_rules", () => {
+    expect(pass).to.changeAST(
+      [
+        "one = three / 'a' / [c-g]",
+        "two = three / 'P' / [Q-T]",
+        "three = 'x' / [u-w]",
```

Comment:
Mostly agreed - I'll certainly add more tests.

> For example, if three rule will contain $ operator, or action block

In that case, its `expression`'s  type will be `text` or `action` and `asClass` will ignore it. So I don't need to do anything to handle that.

But you made me realize that I should explicitly check for `text` and look through it, because we actually want 
`foo = "a" / $("b" / [xyz]) / "X"` to get converted to `foo = [Xabxyz]`.